### PR TITLE
Harry/helpertestfunctions

### PR DESCRIPTION
**Pull request by hwh33**
This is to address Issue #96 and Issue #117.

Adds a package helpers/testsuite which includes functions for generation of certificates and certificate chains in the code (so no more need for pre-made test data).  Also adds the capability to start up a cfssl server running on your local machine.  

These functions all use the CLI in one way or another, so any updates to the way these things are done with the command-line tools will carry-through to this package.

Test coverage for this package is currently at 74%.  All of the uncovered code is very easy to hit, so I will work on bringing it up to 100%.

### CODE REVIEW

File: helpers/testsuite/testing_helpers_test.go [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+		t.Log("The following fields were unequal:")
+		for _, field := range unequalFields {
+			t.Log(field)
+		}
+		t.Fatal("Certificates unequal.")
+	}
+
+}
+
+/* Compare two x509 certificate chains. We only compare relevant data to
```
Comment (kisom) [inline helpers/testsuite/testing_helpers_test.go]: A style nit: we prefer the line commentary style over the C block-comment style.

**Comment by kisom** [comment]
I think the C block-style commentary in `helpers/testsuite/testing_helpers_test.go` should be changed to the C++ line-style commentary (e.g. each line starting with `//`). Other than that, LGTM.

**Comment by hwh33** [comment]
@kisom Ah okay, I will change that then.  Thanks!

**Comment by lziest** [comment]
please fix build failure. Note it is not a coverage faiure.

**Comment by grittygrease** [comment]
Here are two errors that are reported:
api/bundle/bundle.go:55: not enough arguments in call to h.bundler.BundleFromRemote
cli/bundle/bundle.go:78: not enough arguments in call to b.BundleFromRemote

Once these are fixed, please squash and rebase.

**Comment by grittygrease** [comment]
Tests are passing. Please squash and rebase.

**Pull request merged**